### PR TITLE
Add support for OIDC when pushing images from github to AWS ECR

### PR DESCRIPTION
**Pull request by haavardeide**
* Add a new tf/ folder with terraform script to create the OIDC setup in AWS
  * A OIDC provider
  * A role that okctl-hello can assume
  * Re-usable tf module to be able to create multiple roles
* Update release.yaml to use  `aws-actions/configure-aws-credentials` and `aws-actions/amazon-ecr-login` to utilize the OIDC setup

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,31 +1,38 @@
-# This is a basic workflow to help you get started with Actions
+name: Build and deploy main to dev
```
Comment (deifyed) [inline .github/workflows/release.yml:1]: To dev? AFAIK we only have prod for this app

File: tf/modules/iam/github_actions.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    Team = "okctl-hello"
+    Env  = "dev"
+  }
+}
+
+# For thumbprint_list value: see `tf/bin/oidc.sh`
+resource "aws_iam_openid_connect_provider" "github_oidc" {
+  url             = "https://token.actions.githubusercontent.com"
+  client_id_list  = ["sts.amazonaws.com"]
+  thumbprint_list = ["6938fd4d98bab03faadb97b34396831e3780aea1"]
```
Comment (yngvark) [inline tf/modules/iam/github_actions.tf:12]: The current expiry date of the certificate of github.com is "Thursday, March 31, 2022 at 1:59:59 AM", which probably means this github action will stop working at that date. How should we deal with this?

**Review by yngvark — APPROVED** [review]
:100:

File: tf/modules/iam/github_actions.tf (hunk 2 above)
Comment (haavardeide) [reply tf/modules/iam/github_actions.tf:12]: `tf/bin/oidc.sh` will generate a new fingerprint, but that needs to be done manually.
In a perfect world there would be a lambda function doing a validation and looking at expiration, then notifying the user :)
But for now I'll update the documentation a bit more

**Review by deifyed — CHANGES_REQUESTED** [review]
I failed to realise this earlier, okctl-hello doesn't have an IAC repo. This should be moved to the reference app. My bad

File: .github/workflows/release.yml (hunk 1 above)
Comment (haavardeide) [reply .github/workflows/release.yml:1]: @deifyed https://app.okctl-reference-dev.oslo.systems/

**Comment by haavardeide** [comment]
Wrong repo - see okctl-reference-app

**Pull request closed without merging**